captum/attr/_core/layer/layer_ablation.py

```diff
@@ -0,0 +1,273 @@
+                            ablations_per_eval * #examples samples.
+                            For DataParallel models, each batch is split among the
+                            available devices, so evaluations on each available
+                            device contain at most
+                            (ablations_per_eval * #examples) / num_devices
+                            samples.
+                            Default: 1
+
+            Returns:
+                *tensor* of **attributions**:
+                - **attributions** (*tensor*):
+                            Attribution of each neuron in given layer input or
+                            output. Attributions will always be the same size as
+                            the input or output of the given layer, depending on
+                            whether we attribute to the inputs or outputs
+                            of the layer which is decided by the input flag
+                            `attribute_to_layer_input`.
+
+            Examples::
+
+            >>> # SimpleClassifier takes a single input tensor of size Nx4x4,
+            >>> # and returns an Nx3 tensor of class probabilities.
+            >>> # It contains an attribute conv1, which is an instance of nn.conv2d,
+            >>> # and the output of this layer has dimensions Nx12x3x3.
+            >>> net = SimpleClassifier()
+            >>> # Generating random input with size 2 x 4 x 4
+            >>> input = torch.randn(2, 4, 4)
+            >>> # Defining FeatureAblation interpreter
+            >>> ablator = FeatureAblation(net, net.conv1)
+            >>> # Computes ablation attribution, ablating each of the 108
+            >>> # neurons independently.
+            >>> attr = ablator.attribute(input, target=1)
+
+            >>> # Alternatively, we may want to ablate neurons in groups, e.g.
+            >>> # grouping all the neurons in the same row.
+            >>> # This can be done by creating a layer mask as follows, which
+            >>> # defines the neuron groups, e.g.:
+            >>> # +---+---+---+
+            >>> # | 0 | 0 | 0 |
+            >>> # +---+---+---+
+            >>> # | 1 | 1 | 1 |
+            >>> # +---+---+---+
+            >>> # | 2 | 2 | 2 |
+            >>> # +---+---+---+
+            >>> # With this mask, all the 36 neurons in a row / channel are ablated
+            >>> # simultaneously, and the attribution for each neuron in the same
+            >>> # group (0 - 2) per example are the same.
+            >>> # The attributions can be calculated as follows:
+            >>> # layer mask has dimensions 1 x 3 x 3
+            >>> layer_mask = torch.tensor([[[0,0,0],[1,1,1],
+            >>>                             [2,2,2]]])
+            >>> attr = ablator.attribute(input, target=1,
+            >>>                          layer_mask=layer_mask)
+        """
+
+        def layer_forward_func(*args):
```

Comment:
nit: could this be passed in 2 arguments, like ?
```
layer_input =inputs
original_inputs = additional_forward_args
```

Reply:
This would work if layer_input is guaranteed to be 1 tensor / element as it is now, but once that becomes variable, I'm not sure how to cleanly separate it, since _run_forward in feature ablation will concatenate the inputs and additional_args tuples?